addons/account/models/account_tax.py

```diff
@@ -266,28 +266,29 @@ def _check_repartition_lines(self, lines):
         if len(base_line) != 1:
             raise ValidationError(_("Invoice and credit note distribution should each contain exactly one line for the base."))
 
-    @api.constrains('invoice_repartition_line_ids', 'refund_repartition_line_ids')
+    @api.constrains('repartition_line_ids')
```

Comment:
just adding `repartition_line_ids` in the constrains seems to be enough, without the other changes, unless I'm missing something?

Reply:
Actually you don't even need to change anything in the constraint.
I'm not sure if the default lines were correctly computed with the current fix
```diff
diff --git a/addons/account/models/account_tax.py b/addons/account/models/account_tax.py
index ddfb87cb674..8291fba17f1 100644
--- a/addons/account/models/account_tax.py
+++ b/addons/account/models/account_tax.py
@@ -216,8 +216,18 @@ class AccountTax(models.Model):
             fields_list += ['company_id']
         rslt = super().default_get(fields_list)
 
+        company_id = rslt.get('company_id')
+        if 'invoice_repartition_line_ids' in fields_list and 'invoice_repartition_line_ids' not in rslt:
+            rslt['invoice_repartition_line_ids'] = [
+                Command.create({'document_type': 'invoice', 'repartition_type': 'base', 'tag_ids': [], 'company_id': 
company_id}),
+                Command.create({'document_type': 'invoice', 'repartition_type': 'tax', 'tag_ids': [], 'company_id': company_id}),
+            ]
+        if 'refund_repartition_line_ids' in fields_list and 'refund_repartition_line_ids' not in rslt:
+            rslt['refund_repartition_line_ids'] = [
+                Command.create({'document_type': 'refund', 'repartition_type': 'base', 'tag_ids': [], 'company_id': company_id}),
+                Command.create({'document_type': 'refund', 'repartition_type': 'tax', 'tag_ids': [], 'company_id': company_id}),
+            ]
         if 'repartition_line_ids' in fields_list and 'repartition_line_ids' not in rslt:
-            company_id = rslt.get('company_id')
             rslt['repartition_line_ids'] = [
                 Command.create({'document_type': 'invoice', 'repartition_type': 'base', 'tag_ids': [], 'company_id': company_id}),
                 Command.create({'document_type': 'invoice', 'repartition_type': 'tax', 'tag_ids': [], 'company_id': company_id}),
@@ -331,6 +341,8 @@ class AccountTax(models.Model):
         sanitized = vals.copy()
         # Allow to provide invoice_repartition_line_ids and refund_repartition_line_ids by dispatching them
         # correctly in the repartition_line_ids
+        if sanitized.get('invoice_repartition_line_ids') or sanitized.get('refund_repartition_line_ids'):
+            sanitized['repartition_line_ids'] = []
         for doc_type in ('invoice', 'refund'):
             fname = f"{doc_type}_repartition_line_ids"
             if fname in sanitized:
@@ -340,6 +352,7 @@ class AccountTax(models.Model):
                     else (command, id, v)
                     for command, id, v in sanitized.pop(fname)
                 ])
+                sanitized[fname] = []
         return sanitized
 
     @api.model_create_multi
```